installer/Octopus.Tentacle.Installer/Product.wxs

```diff
@@ -1,80 +1,59 @@
-
+    <MajorUpgrade AllowDowngrades="yes" Schedule="afterInstallInitialize" />
     <!--
     Previous install location
     -->
     <Property Id="INSTALLLOCATION">
-      <RegistrySearch Id="RegistrySearch" Type="raw" Root="HKLM"
-                Key="Software\Octopus\Tentacle" Name="InstallLocation" />
+      <RegistrySearch Id="RegistrySearch" Type="raw" Root="HKLM" Key="Software\Octopus\Tentacle" Name="InstallLocation" />
     </Property>
-
     <!--
     Prior to 2.1 our installer was perUser. We changed it to perMachine, but our usual approach of uninstalling the old
     version when upgrading doesn't work when going from perUser to perMachine. Instead, we'll check whether we have this registry key,
     and if we do but we think the product isn't installed, then they must have an old version installed.
     We can't just use the INSTALLLOCATION variable because it gets set once they progress through the wizard.
     -->
     <Property Id="CURRENTLYINSTALLED">
-      <RegistrySearch Id="RegistrySearchForCurrentInstallStatus" Type="raw" Root="HKLM" Win64="$(var.Win64)"
-                Key="Software\Octopus\Tentacle" Name="InstallLocation" />
+      <RegistrySearch Id="RegistrySearchForCurrentInstallStatus" Type="raw" Root="HKLM" Win64="$(var.Win64)" Key="Software\Octopus\Tentacle" Name="InstallLocation" />
     </Property>
     <Condition Message="A previous version of Octopus Deploy Tentacle 2.0 is currently installed. This version cannot be automatically upgraded. You will need to uninstall this version before upgrading. Please view this page for details: http://g.octopushq.com/UninstallFirst">Not CURRENTLYINSTALLED Or WIX_UPGRADE_DETECTED Or Installed</Condition>
-
     <!--
     Ensure that they're not trying to run the 32-bit installer on a 64-bit Operating System.
     -->
     <?if $(var.Platform) = x86 ?>
-    <Condition Message="This installer is only supported on 32-bit operating systems. Please download and run the 64-bit installer.">
-      <![CDATA[Not VersionNT64]]>
-    </Condition>
-    <?endif?>
-
+    <Condition Message="This installer is only supported on 32-bit operating systems. Please download and run the 64-bit installer."><![CDATA[Not VersionNT64]]></Condition>
+    <?endif ?>
     <!--
     Ensure that the .NET Framework has been installed
     -->
-    <PropertyRef Id="NETFRAMEWORK45"/>
-    <PropertyRef Id="NETFRAMEWORK40FULLINSTALLROOTDIR"/>
-    <Condition Message="This application requires Microsoft .NET Framework 4.5 Runtime in order to run. Please install the .NET Framework and then run this installer again.">
-      <![CDATA[Installed OR NETFRAMEWORK45]]>
-    </Condition>
-
+    <PropertyRef Id="NETFRAMEWORK45" />
+    <PropertyRef Id="NETFRAMEWORK40FULLINSTALLROOTDIR" />
+    <Condition Message="This application requires Microsoft .NET Framework 4.5 Runtime in order to run. Please install the .NET Framework and then run this installer again."><![CDATA[Installed OR NETFRAMEWORK45]]></Condition>
```

Comment:
I don't think this message about .NET framework 4.5 is still true

Reply:
🤔 I think this was auto generated. Let me see where it got this information from.

Reply:
Can I please get some help on this? Don't want to dump too much time into learning Wix (can do that later). But from what I can see from [the manual](https://wixtoolset.org/documentation/manual/v3/customactions/wixnetfxextension.html), there's no good replacement for `NETFRAMEWORK45`  for `net48`. This is wix v3. Not sure about v4.

Also found a [snippet in their source code](https://github.com/wixtoolset/wix3/blob/c02e48ec301a60eba88a3b519d47e88eeaa4c978/src/ext/NetFxExtension/wixlib/NetFx48.wxs) that has both similar and very different settings. A little confused...